integration/hibernate-4/src/main/java/com/blazebit/persistence/impl/hibernate/Hibernate4Integrator.java

```diff
@@ -36,7 +37,8 @@ public void integrate(Configuration configuration, SessionFactoryImplementor ses
 				// TODO: check that no collections are mapped
 			}
 		}
-		CommonHibernateUtils.setCollectionPersisterClass(configuration.getClassMappings(), CustomOneToManyPersister.class);
+
+		serviceRegistry.locateServiceBinding(PersisterClassResolver.class).setService(new CustomPersisterClassResolver());
```

Comment:
Honestly I have no idea how to do that properly. If the tests run through, I am fine with any approach.